app/src/main/java/ch/epfl/sweng/SDP/home/HomeActivity.java

```diff
@@ -169,6 +184,40 @@ protected void onCreate(Bundle savedInstanceState) {
         setLeague();
     }
 
+    /**
+     * Checks if storage permissions are granted.
+     * If permissions are revoked it requests permission.
+     * @return a boolean indicating if permissions are granted or not.
+     */
+    public  boolean isStoragePermissionGranted() {
```

Comment:
Pass the sdk version as argument to this method (injection like). This way in the tests you can set a version number and check if the method behaves correctly.